pre_commit_hooks/cfn_guard.py

```diff
@@ -0,0 +1,160 @@
+"""
+This module contains the logic for the cfn-guard pre-commit hook
+"""
+
+import json
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+from pathlib import Path
+from typing import Sequence, Union
+from urllib.request import Request, urlopen
+
+# pylint: disable=C0301
+LATEST_RELEASE_URL = (
+    "https://api.github.com/repos/aws-cloudformation/cloudformation-guard/releases/latest"
+)
+BIN_NAME = "cfn-guard"
+UNSUPPORTED_OS_MSG = "Unsupported operating system. Could not install cfn-guard."
```

Comment:
Another thing to consider is to use bind to the rust library (and publishing wheels to pypi for different OSes). That way you still get a download for supported OSes, but instead of failing on unsupported OSes, customers can install a python/rust toolchain and build from source code

Reply:
That's an interesting approach. This implementation should cover win, mac, and linux so I dropped this error in to see what the customer demand is for other operating systems outside of those three. I like this idea though.